src/DockAreaWidget.cpp

```diff
@@ -745,8 +746,14 @@ void CDockAreaWidget::updateTitleBarVisibility()
 
 	if (d->TitleBar)
 	{
-		bool Hidden = Container->hasTopLevelDockWidget() && (Container->isFloating()
-			|| CDockManager::testConfigFlag(CDockManager::HideSingleCentralWidgetTitleBar));
+		bool Hidden = (
+					Container->hasTopLevelDockWidget() && (Container->isFloating()
+						|| CDockManager::testConfigFlag(CDockManager::HideSingleCentralWidgetTitleBar))
+					)
+				||
+					(
+						d->HideSingleWidgetTitleBar && dockWidgetsCount() == 1

```

Comment:
The function dockWidgetsCount() is not the right choice here and does not work properly. If you close a dock widget with the close button, then dockWidgetsCount() will not change. You should use openDockWidgetsCount() here